localstack/services/s3/models.py

```diff
@@ -48,4 +49,40 @@ class S3Store(BaseStore):
     )
 
 
-s3_stores = AccountRegionBundle("s3", S3Store)
+class BucketCorsIndex:
```

Comment:
I am not completely sure if the `BucketCorsIndex` is current for all scenarios.
- The buckets are cached with the first access to teh `buckets` property (and stay the same as long as they are not invalidated).
- The cache is not invalidated when a bucket is created, only when a bucket is deleted, or cors rules are added or deleted.

I think this could cause a bucket to not be detected in the following scenario:
1. Create a bucket with cors rules.
2. Access the bucket (where the index here is accessed, and populated - at latest).
3. Create a new bucket.
4. Directly afterwards, access the bucket.

Since the cache of the index has not been invalidated, the S3 CORS handler (which uses the index) will not detect the bucket, right?

Reply:
See above, the cache is invalidated in `create_bucket`. 👍